lumicks/pylake/kymotracker/detail/msd_estimation.py

```diff
@@ -447,3 +452,257 @@ def determine_optimal_points(frame_idx, coordinate, max_iterations=100):
         RuntimeWarning("Warning, maximum number of iterations exceeded. Returning best solution.")
     )
     return num_slope, num_intercept
+
+
+def _var_cve_unknown_var(
+    diffusion_constant: float,
+    variance_loc: float,
+    dt: float,
+    num_points: int,
+    blur_constant: float = 0,
+    avg_frame_steps: float = 1,
+) -> float:
+    """Expected variance of the diffusion estimate obtained with CVE.
+
+    The covariance-based diffusion estimator provides a simple unbiased estimator of diffusion.
+    This function is based on eqn 22 from Vestergaard et al [2] adapted for 1D. See also eqn 17 from
+    Vestergaard et al [1].
+
+    Parameters
+    ----------
+    diffusion_constant : float
+        Estimate of the diffusion constant
+    variance_loc : float
+        Estimate of the localization variance
+    dt : float
+        Time step
+    num_points : int
+        Number of points in the trace
+    blur_constant : float
+        Motion blur constant. See :func:`lumicks.pylake.kymotracker.detail.msd_estimation._cve` for
+        more information.
+    avg_frame_steps : float
+        Average frame steps. This number is the average of the time between two localizations in
+        frames mean(delta time between frame / frame timestep). If all frames had successful
+        localization, this constant will be 1.
+
+    References
+    ----------
+    .. [1] Vestergaard, C. L., Blainey, P. C., & Flyvbjerg, H. (2014). Optimal estimation of
+           diffusion coefficients from single-particle trajectories. Physical Review E, 89(2),
+           022726.
+    """
+    epsilon = variance_loc / (diffusion_constant * dt) - 2.0 * blur_constant
+    numerator = 6.0 * avg_frame_steps**2 + 4.0 * epsilon * avg_frame_steps + 2.0 * epsilon**2
+    denominator = num_points * avg_frame_steps**2
+    term1 = numerator / denominator
+    term2 = (4.0 * (avg_frame_steps + epsilon) ** 2) / (num_points**2 * avg_frame_steps**2)
+    return (term1 + term2) * diffusion_constant**2
```

Comment:
I would use built-in functions where applicable (`math.pow()`) - unless there's a reason not to. Perhaps also handle divide by zero scenarios.
```suggestion
    try
        epsilon = variance_loc / (diffusion_constant * dt) - 2.0 * blur_constant
        numerator = 6.0 * pow(avg_frame_steps,2) + 4.0 * epsilon * avg_frame_steps + 2.0 * pow(epsilon, 2)
        denominator = num_points * pow(avg_frame_steps, 2)
        term1 = numerator / denominator
        term2 = (4.0 * pow((avg_frame_steps + epsilon) ,2)) / pow(num_points,2) * pow(avg_frame_steps,2))
        return (term1 + term2) * pow(diffusion_constant,2)
    except ZeroDivisionError:
         raise ValueError("Diffusion constant, blur constant and time step parameters should not be 0")
```

Reply:
Why would I prefer `math.pow` over `**`? Personally I find the operator `**` much more readable.

That's a good point about division by zero. Though, the error message in the division by zero exception you suggest is not correct. The blur constant can definitely be zero (and in our case always is). The diffusion constant can actually also be zero numerically, and this indeed suggests that I should rewrite the equation a little bit.

Time step should not be zero; but I don't think it can be by construction.

Reply:
> Why would I prefer math.pow over **? Personally I find the operator ** much more readable and this code is not exactly performance critical.

Readability was the main thing, and also considering operator `**` has differing meanings based on context one might prefer the distinction. Comes down to preference and the intended audience I suppose. 

> Edit: I've also added a test which specifically includes a test with zero diffusion constant.

Even better!